src/quantization/quantizer_test.h

```diff
@@ -0,0 +1,198 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <catch2/catch_test_macros.hpp>
+
+#include "fixtures.h"
+#include "quantizer.h"
+#include "simd/simd.h"
+using namespace vsag;
+
+template <typename T>
+void
+TestQuantizerEncodeDecode(Quantizer<T>& quant, int64_t dim, int count, float error = 1e-5) {
+    auto vecs = fixtures::generate_vectors(count, dim);
+    quant.ReTrain(vecs.data(), count);
+
+    // Test EncodeOne & DecodeOne
+    auto idx = random() % count;
+    auto* codes = new uint8_t[quant.GetCodeSize()];
+    quant.EncodeOne(vecs.data() + idx * dim, codes);
+    auto* outVec = new float[dim];
+    quant.DecodeOne(codes, outVec);
+    for (int i = 0; i < dim; ++i) {
+        REQUIRE(std::abs(vecs[idx * dim + i] - outVec[i]) < error);
```

Comment:
should replace abs with fabs?

Reply:
For integral arguments, the integral overloads of std::abs are likely better matches. 
reference: https://en.cppreference.com/w/cpp/numeric/math/fabs